internal/generator/vector/outputs.go

```diff
@@ -43,19 +66,28 @@ func Outputs(clspec *logging.CollectionSpec, secrets map[string]*corev1.Secret,
 			}
 		}
 		inputs := ofp[o.Name].List()
-		switch o.Type {
-		case logging.OutputTypeKafka:
-			outputs = generator.MergeElements(outputs, kafka.Conf(o, inputs, secret, op))
-		case logging.OutputTypeLoki:
-			outputs = generator.MergeElements(outputs, loki.Conf(o, inputs, secret, op))
-		case logging.OutputTypeElasticsearch:
-			outputs = generator.MergeElements(outputs, elasticsearch.Conf(o, inputs, secret, op))
-		case logging.OutputTypeCloudwatch:
-			outputs = generator.MergeElements(outputs, cloudwatch.Conf(o, inputs, secret, op))
-		case logging.OutputTypeGoogleCloudLogging:
-			outputs = generator.MergeElements(outputs, gcl.Conf(o, inputs, secret, op))
-		case logging.OutputTypeSplunk:
-			outputs = generator.MergeElements(outputs, splunk.Conf(o, inputs, secret, op))
+
```

Comment:
Normalize so Ignore is handled _exactly_ the same as MaxRecordsPerSecond == 0. 
It is logically equivalent so the same optimizations should apply either way.

```suggestion
if o.Limit.MaxRecordsPerSecond == 0 {
  o.Limit.Policy = logging.IgnorePolicy // Handle limit==0 as Ignore
}
if o.Limit.Policy == logging.Ignore {
  o.Limit.MaxRecordsPerSecond = 0 // make sure we ignore any max value given with Ignore
}

```